src/overlays/actors/ovl_En_Ani/z_en_ani.c

```diff
@@ -28,62 +44,314 @@ const ActorInit En_Ani_InitVars = {
+        this->actionFunc = EnAni_IdleStandingUnused;
+        this->actor.velocity.y = -25.0f;
+        this->actor.minVelocityY = -25.0f;
+        this->actor.gravity = -5.0f;
+    }
+}
+
+void EnAni_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnAni* this = THIS;
+    Collider_DestroyCylinder(globalCtx, &this->collider1);
+    Collider_DestroyCylinder(globalCtx, &this->collider2);
+}
+
+void EnAni_SetText(EnAni* this, GlobalContext* globalCtx, u16 textId) {
+    s16 diffAngle = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+
+    this->actor.textId = textId;
+    if ((this->stateFlags & ANI_STATE_SUFFERING) != 0 || ABS_ALT(diffAngle) < 0x4301) {
+        if (this->actor.xzDistToPlayer < 100.0f) {
+            func_800B8614(&this->actor, globalCtx, 120.0f);
+        }
+    }
+}
+
+/**
+ * just stands there staring at you,
+ * this is NOT him as a shopkeeper, that is EnOsn using his object
+ */
+void EnAni_IdleStandingUnused(EnAni* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+}
+
+void EnAni_Dialogue(EnAni* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+    if (func_80152498(&globalCtx->msgCtx) == 2 && globalCtx->msgCtx.unk11F04 == 0x6DE) {
+        this->actionFunc = EnAni_Suffering;
+    }
+}
+
+/**
+ * writhing, waiting for player to talk to us
+ */
+void EnAni_Suffering(EnAni* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        this->actionFunc = EnAni_Dialogue;
+    } else {
+        // telling you not to take his rupees you knocked from the tree
+        EnAni_SetText(this, globalCtx, 0x6DE);
+    }
+}
+
+/**
+ * waiting for falling over animation to end
+ */
+void EnAni_FallingOver(EnAni* this, GlobalContext* globalCtx) {
+    if (SkelAnime_Update(&this->skelAnime)) {
+        this->blinkFunc = EnAni_WaitForEyeOpen;
+        this->actionFunc = EnAni_Suffering;
+        Animation_Change(&this->skelAnime, &gAniHoldingFootPainWrithingAnim, 1.0f, 0.0f,
+                         Animation_GetLastFrame(&gAniHoldingFootPainWrithingAnim), 0, 0.0f);
+    }
+}
+
+/**
+ * pause: waiting for landing animation and hammer sfx to end
+ */
+void EnAni_WaitingForFootPain(EnAni* this, GlobalContext* globalCtx) {
+    if (SkelAnime_Update(&this->skelAnime)) {
+        this->actionFunc = EnAni_FallingOver;
+        Animation_Change(&this->skelAnime, &gAniFallOverHoldingFootAnim, 1.0f, 0.0f,
+                         Animation_GetLastFrame(&gAniFallOverHoldingFootAnim), 2, 0.0f);
+    }
+}
 
-extern ColliderCylinderInit D_80968670;
-extern InitChainEntry D_8096869C[];
+void EnAni_FallingToGround(EnAni* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s16 quakeValue;
 
-extern UNK_TYPE D_06000C14;
-extern UNK_TYPE D_060011CC;
-extern UNK_TYPE D_060027A0;
-extern UNK_TYPE D_06009220;
-extern UNK_TYPE D_06009D34;
+    // if hit the ground
+    if ((this->actor.bgCheckFlags & 1) != 0) {
```

Comment:
```suggestion
    if (this->actor.bgCheckFlags & 1) {
```
Flag check does not need `!= 0`